src/mmds/src/token.rs

```diff
@@ -0,0 +1,574 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Temporarily disable unused warnings.
+// TODO: remove these once the Token integration is completed.
+#![allow(dead_code)]
+
+use aes_gcm::aead::NewAead;
+use aes_gcm::{AeadInPlace, Aes256Gcm, Key, Nonce};
+use bincode::{DefaultOptions, Options};
+use logger::warn;
+use serde::{Deserialize, Serialize};
+use std::convert::TryInto;
+use std::fs::File;
+use std::io::Read;
+use std::ops::Add;
+use std::path::Path;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use std::{fmt, io};
+
+/// Length of initialization vector.
+pub const IV_LEN: usize = 12;
+/// Length of the key used for encryption.
+pub const KEY_LEN: usize = 32;
+/// Length of encryption payload.
+pub const PAYLOAD_LEN: usize = std::mem::size_of::<u64>();
+/// Length of encryption tag.
+pub const TAG_LEN: usize = 16;
+
+/// Minimum lifetime of token.
+pub const MIN_TOKEN_TTL_SECONDS: u32 = 1;
+/// Maximum lifetime of token.
+pub const MAX_TOKEN_TTL_SECONDS: u32 = 21600;
+
+/// Path to token.
+pub const PATH_TO_TOKEN: &str = "/latest/api/token";
+/// Randomness pool file path.
+const RANDOMNESS_POOL: &str = "/dev/urandom";
+
+/// Token length limit to ensure we don't bother decrypting huge character
+/// sequences. Tokens larger than this are automatically rejected.
+const TOKEN_LENGTH_LIMIT: usize = 100;
+/// Byte limit passed to `bincode` to guard against allocating
+/// too much memory when deserializing tokens.
+const DESERIALIZATION_BYTES_LIMIT: u64 = 50;
```

Comment:
I would add some details about how these values were chosen.

Reply:
I added some additional comments to explain the logic behind these constants. LMKWYT.

Reply:
```
The value is computed
/// based on the size of Token struct, including a small deviation.
```
Can you specify the `size_of(Token)` and maybe provide an approximate formula ?